cabal-install/src/Distribution/Client/Main.hs

```diff
@@ -355,6 +355,28 @@ mainWorker args = do
             warnIfAssertionsAreEnabled
             action globalFlags
   where
+    delegateToExternal :: [Command Action]
+                  -> String
+                  -> [String]
+                  -> IO (CommandParse Action)
+    delegateToExternal commands' name cmdArgs = do
+      mCommand <- findProgramOnSearchPath normal defaultProgramSearchPath ("cabal-" <> name)
```

Comment:
I thought this would fix the test failure on windows but it still seems to persist.